int64       operate
float64   face_dist
float64   z
int64       color